content/news/04-clash122.md

```diff
@@ -0,0 +1,68 @@
+toc: false
+mathjax: false
+---
+
+Hi everyone!
+
+Last Friday we released Clash 1.2.2 on GitHub. In big part thanks to our lovely community, we've found and fixed many bugs. Thanks to some patches, Clash even runs anywhere from 2-20% faster for some common design patterns! Of course, the full list of changes can be found in [the CHANGELOG](https://github.com/clash-lang/clash-compiler/releases/tag/v1.2.2) over at GitHub. With this blogpost I'd like to highlight some of the progress we've made since the 1.2 release, and elaborate on our future plans. 
+
+# Starter project
+With the release of Clash 1.2.2 we've also released the first iteration of our [starter project](https://github.com/clash-lang/clash-compiler/blob/b9cd32cd876eaee1d5591f40c20b1238be195159/clash-starters/clash-example-project/README.md)! The goal of this starter project is to provide a fresh template for a new Clash project. Dependencies can now be added exactly like you would for a Haskell project; simply by adding it to your dependency list in the project's cabal file. Whereas previously you'd have to write custom scripts, or rely on nix to solve it for you!
+
+# Memory leak begone
+For a while now we've seen pretty bad memory leaks in commercial deployments of Clash, but haven't been able to replicate this in a small example. Through various workarounds we've been able to reduce the memory usage to make a usable on "reasonable" hardware and moved on to other, by that time higher priority, issues. We were recently confronted with this issue again, reported on our [public issue tracker](https://github.com/clash-lang/clash-compiler/issues/1256). This time it even had a small example! Time to investigate.
+
+Clash designs operate on infinite streams of data called [signals](https://hackage.haskell.org/package/clash-prelude-1.2.2/docs/Clash-Signal.html). Signals model (bundles of) wires, where each value corresponds to a stabilized signal at the end of a clock cycle. Signals do not always carry known values. For example, in some hardware configurations the very first value produced by a memory element is unknown. For this purpose, Clash exports a bottom called [XException](https://hackage.haskell.org/package/clash-prelude-1.2.2/docs/Clash-XException.html). This presents a problem: for performance reasons you might want to evaluate a value to _Weak Head Normal Form_ (WHNF) or even _Normal Form_ (NF). Usually, you'd use `seq` or `deepseq` to do this, but for typical values used in Clash you can't, as evaluating an `XException` to (WH)NF would stop your circuit's simulation dead in its tracks! To work around this, Clash 1.0 [introduced](/blog/0004-undefined-values/) a type class [NFDataX](https://hackage.haskell.org/package/clash-prelude-1.2.2/docs/Clash-XException.html#t:NFDataX). It allows us to variants of `seq` and `deepseq` that simply ignore `XExceptions`, but otherwise do what you except them to do. Where you'd normally write:
```

Comment:
```suggestion
Clash designs operate on infinite streams of data called [signals](https://hackage.haskell.org/package/clash-prelude-1.2.2/docs/Clash-Signal.html). Signals model (bundles of) wires, where each value corresponds to a stabilized signal at the end of a clock cycle. Signals do not always carry known values. For example, in some hardware configurations the very first value produced by a memory element is unknown. For this purpose, Clash exports a bottom called [XException](https://hackage.haskell.org/package/clash-prelude-1.2.2/docs/Clash-XException.html). This presents a problem: for performance reasons you might want to evaluate a value to _Weak Head Normal Form_ (WHNF) or even _Normal Form_ (NF). Usually, you'd use `seq` or `deepseq` to do this, but for typical values used in Clash you can't, as evaluating an `XException` to (WH)NF would stop your circuit's simulation dead in its tracks! To work around this, Clash 1.0 [introduced](/blog/0004-undefined-values/) a type class [NFDataX](https://hackage.haskell.org/package/clash-prelude-1.2.2/docs/Clash-XException.html#t:NFDataX). It allows us to use variants of `seq` and `deepseq` that simply ignore `XExceptions`, but otherwise do what you except them to do. Where you'd normally write:
```

Reply:
"allows us to variants" -> "allows us to use variants"